pkg/refactor/api/api.go

```diff
@@ -60,8 +69,8 @@ func Start(app core.App, db *database.DB, addr string, done <-chan struct{}, wg
 				regCtr.mountRouter(r)
 			})
 
-			r.Get("/namespaces/{namespace}/gateway_endpoints", func(w http.ResponseWriter, r *http.Request) {
-				writeJSON(w, app.GatewayManager.ListEndpoints())
+			r.Get("/namespaces/{namespace}/endpoints", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
You are introducing three different prefixes:
- /api/v2/gw
- /api/v2/resources/plugins
- /namespaces/{namespace}/endpoints

This is very bad for our api consistency. Also you introduced `resources` segment without proper planning and discussion of what `resources` means.